menuinst/_schema.py

```diff
@@ -36,8 +36,19 @@ class BasePlatformSpecific(BaseModel):
     * Default value is always ``None``.
     """
 
-    name: Optional[constr(min_length=1)] = None
-    "The name of the menu item"
+    name: Optional[
+        Union[
+            constr(min_length=1),
+            Dict[constr(regex=r"^target_environment_is(_not)?_base$"), constr(min_length=1)],
```

Comment:
I think it's easier to just enumerate the two options here with a Literal:

```suggestion
            Dict[Literal["target_environment_is_base", "target_environment_is_not_base"], constr(min_length=1)],
```